### DISCUSSION TITLE
How to write a light curve to a file?

### DISCUSSION THREAD

**Discussion by jrthorstensen** [phoebe-project/phoebe2#483] [General]
I have a nice light curve I'd like to model, and I'm finding I'm flummoxed by a simple procedural issue.  

The tutorials generally end with a nice matplotlib plot, or animation.  What I'd like to do is instead write the modeled light curve to a flle so that I can use other tools to compare it to the data and generate a plot for publication.  I didn't see a clear way to do this -- can anyone enlighten me?  

If it should matter, my usual workflow is not to use notebooks, but rather to write python scripts with a text editor and run them from the command line.

I suppose this could be interpreted as a plea for more elementary documentation!  

Thanks -  John Thorstensen

**Comment by kecnry** [comment]
Hi John,

There is an [export_arrays](http://phoebe-project.org/docs/latest/api/phoebe.parameters.ParameterSet.export_arrays) method for this wrapped around [np.savetxt](https://numpy.org/doc/stable/reference/generated/numpy.savetxt.html).  For example:

```
import phoebe
b = phoebe.default_binary()
b.add_dataset('lc', compute_times=phoebe.linspace(0,1,101))
b.run_compute(model='mymodel')
b.filter(qualifier=['times', 'fluxes'], model='mymodel').export_arrays('mymodel.txt')
```

If that isn't flexible enough for the formatting you need, you can always access the individual arrays in the synthetic model and write them to a file manually:
```
syn_times = b.get_value(qualifier='times', model='mymodel')
syn_fluxes = b.get_value(qualifier='fluxes', model='mymodel')
```

You're right, though, this isn't currently covered in any of the tutorials.  I'll give some thought where to sneak this in so that its easier to find in the future (although surprisingly it doesn't seem that its something many people try to do, so far).  Thanks!

- Kyle

**Reply by jrthorstensen** [reply]
Thank you so much for the quick reply!  That's no doubt just what I needed.  

As for the documentation, I didn't see a comprehensive listing of all the methods in an obvious place, though I may simply have missed it.  Had I found such a listing - which might exist! - I might have been able to find it.

**Reply by kecnry** [reply]
Happy to help!  

All the (public) methods are listed in the [API docs](http://phoebe-project.org/docs/latest/api), but I'll admit you need to do a little digging to find this one since its not very commonly used or advertised (its under "filtering and accessing parameters" or in the complete listing of all methods for [ParameterSet](http://phoebe-project.org/docs/latest/api/phoebe.parameters.ParameterSet) which is linked to at the bottom of the API index page).  But also feel free to post here if you can't find anything!

**Comment by he199219921992** [comment]
hellow sir how i can inpute data in phoebe program version 0.30

**Reply by kecnry** [reply]
If you're looking to input data into the phoebe 0.30 legacy GUI ([website](http://phoebe-project.org/1.0), [github](https://github.com/phoebe-project/phoebe1)), then you can just click the "add" buttons on the "data" panel.  For more details, see the [phoebe legacy user manual](http://phoebe-project.org/static/legacy/docs/phoebe_manual.pdf) (pages 9-17).

To input a model from phoebe 0.30 (legacy) into phoebe 2.x in python, you can use [b.from_legacy](http://phoebe-project.org/docs/latest/api/phoebe.frontend.bundle.Bundle.from_legacy.md).

**Comment by wpr1820** [comment]
Hi,sir.I am new here.How can I input kepler's data into the phoebe 2.4 ? It would be better if there was an example.

**Reply by kecnry** [reply]
PHOEBE does not parse input files as there are many different formats and dedicated packages exist that can help you parse the files.  You'll likely want to use something like numpy's loadtxt or genfromtxt (if loading from an ascii file) or [astropy's fits package](https://docs.astropy.org/en/stable/io/fits/index.html) (if loading from a fits file) to extract the arrays of your observations (times, fluxes, uncertainties).  Once you have the arrays, the [datasets tutorial](http://phoebe-project.org/docs/latest/tutorials/datasets) probably has the most comprehensive examples for loading them into PHOEBE datasets.

**Comment by jvillasr** [comment]
Is it possible to directly obtain/export the phases instead of times? At least naively changing 'times' by 'phases' or adding x='phases' doesn't work for me.

PS: Please let me know if it's better to open a new discussion instead of commenting on old ones.

**Reply by kecnry** [reply]
phases are not exposed in a parameter, but are rather computed on-the-fly from the times and the current ephemeris when requested (see the ["dealing with phases" section in this tutorial](http://phoebe-project.org/docs/latest/tutorials/datasets_advanced) for more details).  To access the phases for any exposed time array, you can call [b.to_phase](http://phoebe-project.org/docs/latest/api/phoebe.frontend.bundle.Bundle.to_phase) and pass the time array.  Unfortunately this means that you'll probably need to write to the file manually (using something like `np.savetxt`) instead of using the [export_arrays ](http://phoebe-project.org/docs/latest/api/phoebe.parameters.ParameterSet.export_arrays) helper method. 

-Kyle

**Reply by jvillasr** [reply]
It worked perfectly, thanks!
